Testing/SystemTests/tests/framework/InterfaceDocumentationTest.py

```diff
@@ -0,0 +1,54 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2023 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+
+from typing import Dict, List
+
+from DocumentationTest import path_to_docs, DocumentationTestBase
+from mantid.kernel import ConfigService
+from workbench.utils.gather_interfaces import gather_interface_names
+from mantidqt.interfacemanager import InterfaceManager
+
+INTERFACES_DOCS_PATH = path_to_docs / "interfaces"
+
+# This should be empty
+INTERFACES_WITH_NO_DOCS = [
+    "Shiver.rst",  # externally imported
+    "SNAPRed.rst",  # externally imported
+    "ORNL SANS.rst",  # information is in concepts/ORNL_SANS_Reduction.rst
+    "Powder Diffraction Reduction.rst",  # reduction application
+    "DGS Reduction.rst",  # reduction application
+    "Direct Data Analysis.rst",  # opens Inelastic Data Analysis
+    "Indirect Data Analysis.rst",  # opens Inelastic Data Analysis
+    "HFIR 4Circle Reduction.rst",
+]
+
+
+def _get_interface_names() -> Dict[str, List[str]]:
+    # need to initialise this so cpp interfaces are registered
+    InterfaceManager()
```

Comment:
Would this line be better off in the `gather_interface_names` utility function, so that we do not need to remember to initialize the InterfaceManager whenever collecting interface names?